### PR TITLE
update golang and dependencies

### PR DESCRIPTION
**Pull request by dallemon**
this pr refactors dockerfile, updates golang to 1.19 and fixes broken packages related to the update.

### CODE REVIEW

**Comment by tcolgate** [comment]
The Dockerfile changes seem largely cosmetic, and I'd rather not fix the alpine version (since that then requires explicit maintenance for security fixes for alpine.

**Comment by dallemon** [comment]
made the requested change.

File: Dockerfile [hunk 1]
```diff
@@ -1,11 +1,14 @@
-#syntax=docker/dockerfile:1.4.2
+#syntax=docker/dockerfile:1.5.1
 
-FROM golang:1.19-alpine AS build
-WORKDIR /go/src/
+FROM cgr.dev/chainguard/go:latest AS build
```
Comment (tcolgate) [inline Dockerfile:3]: I'm not aware of chainguard, and would rather stick to the bog standard docker hub images (or a trustworthy mirror there of)

**Comment by tcolgate** [comment]
Personally I'd probably rather go with klog than logrus for logging. Any specific reason you opted for logrus?

**Comment by dallemon** [comment]
> Personally I'd probably rather go with klog than logrus for logging. Any specific reason you opted for logrus?

It was already in use in the project. :)

**Comment by tcolgate** [comment]
Thanks, and sorry for the slow review.

**Pull request merged**